### PR TITLE
Sub: Fix SET_POSITION_TARGET_LOCAL_NED to be above origin

### PR DESCRIPTION
**Pull request by hendjoshsr71**
Fixes SET_POSITION_TARGET_LOCAL_NED  to be above origin as is the standard . https://mavlink.io/en/messages/common.html#MAV_FRAME_LOCAL_NED

Also removes some redundant code to use functions that exist in AP_Location

### CODE REVIEW

File: ArduSub/control_auto.cpp [hunk 1]
```diff
@@ -463,10 +463,11 @@ void Sub::set_auto_yaw_roi(const Location &roi_location)
 #endif  // HAL_MOUNT_ENABLED
     } else {
 #if HAL_MOUNT_ENABLED
-        // check if mount type requires us to rotate the quad
+        // check if mount type requires us to rotate the sub
         if (!camera_mount.has_pan_control()) {
-            roi_WP = pv_location_to_vector(roi_location);
-            set_auto_yaw_mode(AUTO_YAW_ROI);
+            if(roi_location.get_vector_from_origin_NEU(roi_WP)) {
```
Comment (rmackay9) [inline ArduSub/control_auto.cpp:468]: nitpick: normally we have a space between the "if" and the bracket.

**Comment by rmackay9** [comment]
Yes, we replaced these "pv_" (Position Vector) methods a long time ago in Copter so it makes sense to do the same in Sub.  You've tested it of course?

File: ArduSub/control_auto.cpp (hunk 1 above)
Comment (hendjoshsr71) [reply ArduSub/control_auto.cpp]: done

**Comment by hendjoshsr71** [comment]
> Yes, we replaced these "pv_" (Position Vector) methods a long time ago in Copter so it makes sense to do the same in Sub. You've tested it of course?

Yes, still circles and points at ROIs.

**Review by Williangalvani — APPROVED** [review]

**Pull request merged**